src/Dfc.CourseDirectory.Functions/FixCorruptStatusData.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.Core.Models;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace Dfc.CourseDirectory.Functions
+{
+    /// <summary>
+    /// In prod CourseStatus has got out of sync with the CourseRuns
+    /// </summary>
+    public class FixCorruptStatusData
+    {
+        private readonly Configuration _configuration;
+        private readonly ILogger _logger;
+        private readonly DocumentClient _documentClient;
+
+        public FixCorruptStatusData(
+            Configuration configuration,
+            ILoggerFactory loggerFactory,
+            DocumentClient documentClient)
+        {
+            _configuration = configuration;
+            _documentClient = documentClient;
+            _logger = loggerFactory.CreateLogger<FixCorruptStatusData>();
+        }
+
+        [FunctionName(nameof(FixCorruptStatusData))]
+        [NoAutomaticTrigger]
+        public async Task Run(string input)
+        {
+            var collectionUri = UriFactory.CreateDocumentCollectionUri(
+                _configuration.DatabaseId,
+                _configuration.CoursesCollectionName);
+
+            const CourseStatus courseRunStatus = CourseStatus.Live;
+            const CourseStatus unwantedCourseStatus = CourseStatus.BulkUploadReadyToGoLive;
+
+            var query = _documentClient
+                .CreateDocumentQuery<Course>(collectionUri,
+                    new FeedOptions() {EnableCrossPartitionQuery = true})
+                .Where(c => c.CourseStatus == unwantedCourseStatus)
+                .AsDocumentQuery();
+
+            int fixCount = 0;
+            await query.ProcessAll(async courses =>
+            {
+                foreach (var course in courses.Where(c => c.CourseRuns.All(r => r.RecordStatus == courseRunStatus)))
```

Comment:
Can you not combine this with the initial query?

Reply:
I tried, cosmos linq didn't support what I tried (`All()`), runtime failure
```
[2020-11-04T17:19:24.988] Executed 'FixCorruptStatusData' (Failed, Id=ad67db82-766b-4395-963c-79931c8554b8, Duration=10662ms)
[2020-11-04T17:19:24.988] System.Private.CoreLib: Exception while executing function: FixCorruptStatusData. Microsoft.Azure.DocumentDB.Core: Method 'All' is not supported., Linux/19.3 documentdb-netcore-sdk/2.9.2.
```
Gave up. It's not a huge amount of data after the first filter.